backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/ProofManager.java

```diff
@@ -136,6 +166,93 @@ public void sendPresentProofRequest(@NonNull UUID partnerId, @NonNull RequestPro
         }
     }
 
+    // respond to proof request with valid proof
+    public void presentProof(PresentationExchangeRecord presentationExchangeRecord) {
+        //verify correct state = `request_received`
+        assert presentationExchangeRecord.getState() == PresentationExchangeState.REQUEST_RECEIVED;
+        //find vc's in wallet that satisfy proof
+        Optional<List<PresentationRequestCredentials>> validCredentials = Optional.empty();
+
+        try {
+            validCredentials =  ac.presentProofRecordsCredentials(presentationExchangeRecord.getPresentationExchangeId());
```

Comment:
I'm not sure I understand this. Generally I thought the idea is that when you receive a proof request, you would stop here, send a message to the user and the user then selects what to disclose. The code below looks like an auto respond. This makes sense if it's tied into an auto respond rule. But before starting such work I would generally think about the expected proof formats that you want to handle and write an abstraction for that. This abstraction can then be tested separately and later tied into an auto respond rule.  Otherwise the code will be hard to test and very fragile. An other option would be to simply pipe the whole json through to the UI and do not care at all about semantics here. But then you can not apply any rules of course.

So generally I would propose.

1. think about what you want to achieve. What type of proof requests do you want to handle
2. How should the UI look like
3. Write an handler/abstraction for that
4. For everything else fall back to auto respond. Or if this is not enough fall back to an all or nothing approach in the UI. Where the user sees the whole request but can only approve or deny. 

I think like this the story becomes more manageable.

Reply:
I wanted to ignore the UX/UI requirements of 'holding' the proof requests for a user to intercept until after i understood the construction. That will be changed in this branch or a follow up. 

I'll see what i can do about a more OO approach instead of one procedural method, I agree we will endup with more testable and robust code, it's just not a style am super familiar with.